engine/include/cubos/engine/audio/audio.hpp

```diff
@@ -0,0 +1,27 @@
+/// @file
+/// @brief Class @ref cubos::engine::Audio.
+/// @ingroup audio-plugin
+#pragma once
+
+#include <cubos/core/al/miniaudio_context.hpp>
+#include <cubos/core/memory/stream.hpp>
+#include <cubos/core/reflection/reflect.hpp>
+
+#include <cubos/engine/assets/asset.hpp>
+
+namespace cubos::engine
+{
+    /// @brief Asset containing raw Audio data.
+    ///
+    /// @ingroup audio-plugin
+    struct CUBOS_ENGINE_API Audio
+    {
+        CUBOS_REFLECT;
+        std::shared_ptr<cubos::core::al::impl::Buffer> mData; // Raw data of the audio
+        size_t mLength;                                       // Audio length in seconds TODO: add getter in audio
```

Comment:
```suggestion
        /// @brief Audio buffer.
        cubos::core::al::Buffer buffer;
```
Isn't the length already a property of the buffer?